File: 3/3.1.py
import math

def f(x: float) -> float:
    return (math.sin(x) / (math.log(x) ** 2)) + (math.cos(x) / (math.log(abs(x))))


print(f"x\t|y")
print("+-----------------------------+")

# Способ 1
for i in range(2, 3):
    for j in range(10):
        num = round((i * 10 + j) / 10, 1)
        print(f"{num}\t|{f(num)}")

num += 0.1
print(f"{num}\t|{f(num)}")

print("+-----------------------------+")

# Способ 2
i = 2
while i < 3.1:
    num = round(i, 1)
    print(f"{num}\t|{f(num)}")
    i += 0.1

File: 3/3.1.pas
program f1;
var x, num: real;
var i, j: integer;

function f (x: real): real;
  begin
    Result := (sin(x) / ln(Power(x, 2))) + (cos(x) / ln(Abs(x)))
  end;

begin
writeln('x', #9, '|', 'y');
writeln('+-----------------------------+');
  
//Способ 1
for i := 2 to 2 do
begin
  for j := 0 to 10 do
  begin
    num := (i * 10 + j) / 10;
    writeln(num, #9, '|', f(num))
  end;
end;

writeln('---------------');

//Способ 2
x := 2;
while x < 3.1 do
begin
  writeln(x, #9, '|', f(x));
  x += 0.1;
end;

writeln('---------------');

//Способ 3
x := 2;
repeat
  writeln(x, #9, '|', f(x));
  x += 0.1;
until x > 3.1;

end.